### PR TITLE
Add section about memory safety to `ffi::CString` documentation

### PR DESCRIPTION
**Pull request by dirk**
Also a minor language tweak to the documentation of the `ffi::CString::from_raw` function.

### CODE REVIEW

File: src/libstd/ffi/c_str.rs [hunk 1]
```diff
@@ -60,6 +60,18 @@ use vec::Vec;
 /// }
 /// # }
 /// ```
+///
+/// # Safety
+///
+/// CStrings are intended for working with traditional C-style strings
```
Comment (steveklabnik) [inline src/libstd/ffi/c_str.rs]: Could you put `CString` is graves, both here and below?

Comment (dirk) [inline src/libstd/ffi/c_str.rs]: @steveklabnik: Fixed in 71b526bf944c04ed0ac7d96b898332b4697a53a6!

**Comment by dirk** [comment]
@steveklabnik: Updated docs to have `CString` always be in backticks. Look good to you now?

File: src/libstd/ffi/c_str.rs [hunk 2]
```diff
@@ -209,9 +221,9 @@ impl CString {
 
     /// Retakes ownership of a CString that was transferred to C.
     ///
-    /// The only appropriate argument is a pointer obtained by calling
-    /// `into_raw`. The length of the string will be recalculated
-    /// using the pointer.
+    /// This should only ever be called with a pointer that was earlier
+    /// obtained by calling `into_raw` on a CString. Additionally, the length
```
Comment (steveklabnik) [inline src/libstd/ffi/c_str.rs]: one last `CString` here, then we're good to go :)

Comment (dirk) [inline src/libstd/ffi/c_str.rs]: Oh jeez, I'm hopeless. :disappointed:

**Comment by dirk** [comment]
@steveklabnik: Those last two backtick formatting nits are fixed in 9cfa1916fa06fa5fa809d6d7807a6f978e35c67d.

**Comment by steveklabnik** [comment]
@bors: r+ rollup

Thanks! And no, you're not: I mess these things up all the time :)

**Pull request merged**